### ISSUE TITLE
Bug: cypress-fail-fast does not totally stop testing after finding failure

### ISSUE THREAD

**Issue by jeff00seattle** [javierbrea/cypress-fail-fast#42]
### Problem
If `cypress/integrations` has multiple specs, `cypress-fail-fast` does not total stop all future testing after failing test. 

### Cypress Integrations with multiple Specs
Within my Cypress integrations folder has 20+ specs:

```
cypress/integration
└── service
    └── actions
        ├── stage_0_config
        │   └── cypress.config.spec.ts
        ├── stage_1_admin
        │   ├── admin.0.login.csrf.spec.ts
        │   ├── admin.1.verify.spec.ts
        │   └── admin.2.act.enabled.verify.spec.ts
       ....
```

### Example forced test failure

To checkout `cypress-fail-fast`,  within first spec `cypress.config.spec.ts` added the follow hard-coded failure:

```
  it('TEST Expect Fail Fast', done => {
    done(new Error('Expect Fail Fast'));
  });
```

#### First Spec with hard coded test failure fast failed, as expected
During test run, when it hit the hard-coded failure, all future tests within that spec were skipped, as expected  😃 

```
┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Tests:        12                                                                               │
  │ Passing:      1                                                                                │
  │ Failing:      1                                                                                │
  │ Pending:      0                                                                                │
  │ Skipped:      10                                                                               │
  │ Screenshots:  3                                                                                │
  │ Video:        true                                                                             │
  │ Duration:     0 seconds                                                                        │
  │ Estimated:    0 seconds                                                                        │
  │ Spec Ran:     service/actions/stage_0_config/cypress.config.spec.ts                            │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘
```

#### Next Specs: did run first test for every Spec, even though console log shows skipped
 So, I had expected that all future tests within all future specs under `cypress/integrations` would also be skipped.

However, what happened, the first test of every spec ran, but console shows that they were skipped.

Example of one of the Spec's showing first test ran:

![Example of First Test running in Spec](https://user-images.githubusercontent.com/833269/103426410-33a6fe80-4b6e-11eb-802d-47f68b9a63de.png)

However, console shows that they were skipped:
  ```
┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Tests:        7                                                                                │
  │ Passing:      0                                                                                │
  │ Failing:      0                                                                                │
  │ Pending:      0                                                                                │
  │ Skipped:      7                                                                                │
  │ Screenshots:  0                                                                                │
  │ Video:        true                                                                             │
  │ Duration:     0 seconds                                                                        │
  │ Estimated:    16 seconds                                                                       │
  │ Spec Ran:     service/actions/stage_1_admin/admin.0.login.csrf.spec.ts           │  
└────────────────────────────────────────────────────────────────────────────────────────────────┘
```

**Comment by jeff00seattle** [comment]
Continuing with issue, these are snapshots from Cypress.io dashboard.

First snapshot is when test fails in first spec, and even though first test of each following spec(s) are run (successfully), those specs are considered as skipped because all tests of these specs are considered as skipped:
![cypress_fail_fast_skipped_specs](https://user-images.githubusercontent.com/833269/103433283-82bb5680-4ba3-11eb-9beb-3d688e5ba484.png)

Second snapshot is when test fails in second spec, however the first test of each following spec(s) is run (but failed), those specs are considered as failed even though all other tests of these specs were skipped:
![cypress_fail_fast_skipped_specs_before_failed_or_first_test_ran_failed](https://user-images.githubusercontent.com/833269/103433321-05441600-4ba4-11eb-92b0-caa1b9453a56.png)

**Comment by javierbrea** [comment]
> Second snapshot is when test fails in second spec, however the first test of each following spec(s) is run (but failed), those specs are considered as failed even though all other tests of these specs were skipped:

I think it is a different problem from the one described in this issue. Could you please open a different issue for this comment @jeff00seattle ? Thank you 🙂

**Comment by javierbrea** [comment]
Are you sure about every first test in the rest of specs is really executed? Or is it a problem related to Cypress UI showing it as executed, but is is being skipped? I mean, to what we should believe? To the logs or to the UI?
Could you please add some traces or even force those first tests to fail in order to check if their code is being executed?

**Comment by javierbrea** [comment]
@jeff00seattle, could you try with new release [v2.2.0](https://github.com/javierbrea/cypress-fail-fast/releases/tag/v2.2.0)?

**Comment by javierbrea** [comment]
Hi @jeff00seattle,

Any news on this? Is the issue already happening in the latest version of the plugin? Could be this problem related to the [execution of the `before` hooks](https://github.com/javierbrea/cypress-fail-fast/issues/52), which was already fixed in the v2.2.0 release?

If no more comments are received, I will close the issue due to inactivity. Thanks for your feedback anyway! 🙂

**Comment by javierbrea** [comment]
Closed by inactivity
